### ISSUE TITLE
LUA script not working on Express LRS

### ISSUE THREAD

**Issue by julioghigi** [flyduino/kissfc-tx-lua-scripts#53]
I am using Express LRS with KISS and everything works fine, except by the LUA script. 
It won't load any number, just stay on "----".
I took the same rx and put on a BF board and the BF LUA script is working fine, so I know thta the problem isn't the rx or radio system, just the LUA acript for kiss.
Hope someone can solve this or at least tell me how to solve.
Thanks in advance.

**Comment by fedorcomander** [comment]
What protocol is used by LRS? Crossfire? Sport?

**Comment by julioghigi** [comment]
ELRS uses Crossfire protocol!

**Comment by fedorcomander** [comment]
then it should work. if its STRICTLY follows protocol.

**Comment by julioghigi** [comment]
Yeah I know, but it won't, BF lua script works normally, is there a way I can debug it? Like see what it is receiving and more?

**Comment by fedorcomander** [comment]
u can see in lua itself what is sent, and compare it with bf. ;) may be addresses are not ok?

**Comment by sandrosov85** [comment]
Hello. Give me some help please! What address i shoud use or where i can get it?

If i right understand i need right value in this place:

`LOCAL_DEVICE_ID = 0xEA -- transmitter
REMOTE_DEVICE_ID = 0xC8 -- flight controller`

Please confirm.

**Comment by fedorcomander** [comment]
I have LRS system on the way to me, will check when its here. Otherwise all is 
done based on PDF from trappy.

**Comment by julioghigi** [comment]
Can you please provide me this PDF, I would love to try debug why LUA script don't work and try something.

**Comment by sandrosov85** [comment]
This would be great...

**Comment by fedorcomander** [comment]
no idea. will debug when its here. 

---
Cheers,
Alex

**Comment by julioghigi** [comment]
No idea if you can provide the PDF?

**Comment by fedorcomander** [comment]
ask trappy….

**Comment by sandrosov85** [comment]
Hello Alex! Thanks for your help. Can you send contact @trappy at somewhere?

I try to analize lua from BF and KISS.  Can you explain about adresses?

Found in BF - betaflight-tx-lua-scripts/src/SCRIPTS/BF/MSP/crsf.lua

> -- CRSF Devices
> local CRSF_ADDRESS_BETAFLIGHT          = 0xC8
> local CRSF_ADDRESS_RADIO_TRANSMITTER   = 0xEA
> -- CRSF Frame Types
> local CRSF_FRAMETYPE_MSP_REQ           = **0x7A**      -- response request using msp sequence as command
> local CRSF_FRAMETYPE_MSP_RESP          = **0x7B**      -- reply with 60 byte chunked binary
> local CRSF_FRAMETYPE_MSP_WRITE         = **0x7C**      -- write with 60 byte chunked binary 

And found in kissfc-tx-lua-scripts/src/common/KissProtocolCF.lua

> --
> -- KISS/CF code
> -- Kiss version by Alex Fedorov aka FedorComander
> local toolName = "TNS|KISS Crossfire|TNE"
> LOCAL_DEVICE_ID = 0xEA -- transmitter
> REMOTE_DEVICE_ID = 0xC8 -- flight controller
> FC_REQUEST_COMMAND = **0x78**
> FC_RESPONSE_COMMAND = **0x79**

FC_REQ & FC_RESP are different. Did you speak [at ](https://github.com/flyduino/kissfc-tx-lua-scripts/issues/53#issuecomment-841678859) about this adresses? This addresses needs to be edited for try to fix script???

And 1 more question about settings from pages (like PIDS).
From bf i cant find addresses for ret and set values for pid setting. 
But in kissCF.lua in pages (like PIDS) i see different addresses in start of file. Example:

kissfc-tx-lua-scripts/src/128x64/KISS/pids.lua
>     read = 0x43,
>     write = 0x44,

Is all setting write at different addresses or it is way to make payload to sent to FC?

**Comment by fedorcomander** [comment]
look him up on tbs facebook group. he lives there. 

---
Cheers,
Alex

**Comment by fedorcomander** [comment]
i will take a look when i have hw as i said earlier. sorry. dont have tome now to explain how all this works. 

---
Cheers,
Alex

**Comment by sandrosov85** [comment]
Many thanks!
